src/components/ingestion/AbstractIngestionEngine.js

```diff
@@ -1,100 +1,209 @@
 import { spawn } from 'child_process'
+import {
+	pipeline,
+	Writable,
+	Transform,
+} from 'stream'
 import { Payload } from '@tvkitchen/base-classes'
 import { dataTypes } from '@tvkitchen/base-constants'
 import {
 	AbstractInstantiationError,
 	NotImplementedError,
 } from '@tvkitchen/base-errors'
+import {
+	TSDemuxer,
+} from 'ts-demuxer'
+
 import logger from '%src/lib/logger'
 import kafka from '%src/lib/kafka'
+import {
+	tsToMilliseconds,
+	generateEmptyPacket,
+} from '%src/lib/utils/mpegts'
 
+/**
+ * The AbstractIngestionEngine handles the bulk of the stream processing associated with video
+ * ingestion. Ingeston Engines that implement this class are responsible for defining the input
+ * stream, and the abstract class will handle coordinating the input pipeline that ultimately
+ * produces messages to the STREAM.CONTAINER kafka queue.
+ *
+ * The pipeline is consists of the following steps:
+ *
+ * 1. The input stream emits video data.
+ * 2. That video data is piped to an FFmpeg spawn process in order to wrap it in an MPEG-TS
+ *    container.
```

Comment:
This may not matter, but, these steps are not strictly true. The input stream will almost certainly not be emitting raw video data (implied by the language), but some other container format (e.g., MP4). This first FFmpeg process demuxes/remuxes it into MPEG-TS for consistency. It also providing the point at which (if we ever want) we would normalize audio/video codecs.*

I don't think you have to capture all of that, but at least changing "wrap it in an MPEG-TS container" to "ensure the container format is MPEG-TS" (since I would guess that FFmpeg doesn't perform the demux/remux if the container is already MPEG-TS, but that's just a guess).

*ACTUALLY we currently _are_ changing the format by not configuring FFmpeg otherwise. Check this and I'll bring it up elsewhere (since this is important), but by not explicitly passing in the codec pass-through params to FFmpeg, it's almost certainly re-encoding the input to FFmpeg defaults. (Yes, the default FFmpeg behavior in the absence of being told otherwise is to normalize encoding, not pass streams through unmodified.)

Reply:
Good call; will update!